cardano-node/src/Cardano/Node/Protocol/Shelley.hs

```diff
@@ -165,10 +169,9 @@ readLeaderCredentialsSingleton
      } =
      fmap (:[]) $
      mkPraosLeaderCredentials
-       <$> firstExceptT FileError (newExceptT $ readFileTextEnvelope AsOperationalCertificate certFile)
-       <*> firstExceptT FileError (newExceptT $ readFileTextEnvelope (AsSigningKey AsVrfKey) vrfFile)
-       <*> firstExceptT FileError (newExceptT $ readFileTextEnvelope (AsSigningKey AsKesKey) kesFile)
--- But not OK to supply some of the files without the others.
+       <$> firstExceptT FileError (newExceptT $ readAndDeserialiseFromTextEnvelopeFile AsOperationalCertificate certFile)
+       <*> firstExceptT FileError (newExceptT $ readAndDeserialiseFromFile (AsSigningKey AsVrfKey) (NE.fromList [InputFormatBech32, InputFormatHex, InputFormatTextEnvelope]) vrfFile)
+       <*> firstExceptT FileError (newExceptT $ readAndDeserialiseFromFile (AsSigningKey AsKesKey) (NE.fromList [InputFormatBech32, InputFormatHex, InputFormatTextEnvelope]) kesFile)
```

Comment:
Note how we just use `NE.fromList` so, we did not gain any safety :grin: 

We don't need `InputFormatHex` here right? We're moving from only the text envelope format to accepting bech32 while keeping backwards compat. We don't need to also support hex.

Also, can we pull the `firstExceptT FileError` out to the top rather than repeating?